core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -240,6 +244,7 @@ private[spark] class ExecutorAllocationManager(
     removeTimes.retain { case (executorId, expireTime) =>
       val expired = now >= expireTime
       if (expired) {
+        numTargetExecutorAdjustable.compareAndSet(false, true)
```

Comment:
These `compareAndSet` can just be `set` I think. In fact a `@volatile boolean` is OK too. I don't know what's simpler.